rsts/howto/performance/index.rst

```diff
@@ -4,6 +4,90 @@
 How do I optimize performance of my Flyte Deployment?
 ######################################################
 
-.. CAUTION::
+.. tip:: Before getting started, it is always important to measure the performance. Flyte project publishes and manages some grafana templates as described in - :ref:`howto-monitoring`.
 
-    Coming soon 🛠
+Scaling up FlytePropeller
+==========================
+`FlytePropeller <https://pkg.go.dev/github.com/flyteorg/flytepropeller>`_ is the core engine of Flyte that executes the workflows for Flyte. It is implemented as a `controller <https://kubernetes.io/docs/concepts/architecture/controller/>`_ in Kubernetes.
+FlytePropeller has a lot of knobs that can be tweaked for performance. The default configuration is good enough for small to medium sized installations of Flyte, that are running about 500 workflows concurrently with no noticeable overhead. In the case when the number of workflows increases,
+FlytePropeller will automatically slow down, without losing correctness.
+
+Here are signs of slowdown,
+
+#. Round Latency for each workflow increases
+#. Transition latency increases
+#. Workflows take longer to start
+
+This is usually because the number of threads in FlytePropeller are not enough to keep up with the number of workflows. This can be resolved by adjusting FlytePropeller config specified `here <https://pkg.go.dev/github.com/flyteorg/flytepropeller@v0.10.3/pkg/controller/config>`_.
+
+.. list-table:: Important Properties
```

Comment:
I think for every (most of) property here, we should link to a chart that shows the effect?

Reply:
how to link to a chart?

Reply:
maybe like [this](https://github.com/flyteorg/flyte/blob/master/stats/flytepropeller_dashboard.py#L51-L63)